cmd/thanos/receive.go

```diff
@@ -63,8 +65,7 @@ func registerReceive(app *extkingpin.App) {
 
 	retention := modelDuration(cmd.Flag("tsdb.retention", "How long to retain raw samples on local storage. 0d - disables this retention.").Default("15d"))
 
-	hashringsFile := cmd.Flag("receive.hashrings-file", "Path to file that contains the hashring configuration.").
-		PlaceHolder("<path>").String()
+	hashringsFile := extflag.RegisterPathOrContent(cmd, "receive.hashrings", "JSON file that contains the hashring configuration.", false)
```

Comment:
nit: I don't think we _have_ to specify JSON here, this may turn into YAML in the future anyways